ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +360,709 @@ const createSecret = async (res, appid, payload, DB) => {
+ *                  Object ==> {r}versionId(int) : The version number of the secret value. eg. 12
+ *                  Object ==> {r}createTime(long) : The time when the secret was created, millisecond unit. eg. 1659519772925
+ */
+const listSecretVersionIds = async (res, appid, DB, payload) => {
+    try {
+        // get and check parameter 
+        let secretName = getParam_String(payload, 'secretName')
+        if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+
+        // search secret_version_data, Maximum 4000 line 
+        const query = {
+            selector: {
+                appid: appid,
+                secretName: secretName,
+                deletedFlag: false
+            },
+            fields: ['versionId', 'createTime'],
+            limit: 4000,
+        }
+        let secret_version_data_result = await DB.partitionedFind('secret_version_data', query)
+
+        // build result
+        let result = {
+            'secretName': base64_decode(secretName),
+            'totalCount': 0,
+            'versionIds': []
+        }
+        if (secret_version_data_result.docs.length > 0) {
+            result.totalCount = secret_version_data_result.docs.length
+            for (const doc of secret_version_data_result.docs) {
+                let secretVersion = {};
+                secretVersion['versionId'] = doc['versionId']
+                secretVersion['createTime'] = doc['createTime']
+                result.versionIds.push(secretVersion)
+            }
+        }
+
+        res.send(_result(200, 'List secret versionIds success.', result))
+        return
+    } catch (e) {
+        console.info('listSecretVersionIds :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * Queries all secrets created by your appid. Maximum 4000 line 
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controllerler
+ * @param {Object} payload
+ *          ==> {o}secretName(String [1~64]) : The name of the secret. eg. 'secretName01'
+ * @returns {Object}
+ *          ==> {r}totalCount(int) : The number of returned secrets. eg. 15
+ *          ==> {r}secretList(Array[Object]) : The list of secrets.
+ *                  Object ==> {r}secretName(String) : The name of the secret. eg. 'secretName01'
+ *                  Object ==> {r}description(String) : The description of the secret. eg. 'desc01'
+ *                  Object ==> {r}createTime(long) : The time when the secret was created, millisecond unit. eg. 1659519772925
+ *                  Object ==> {o}plannedDeleteTime(long) : The time when the secret is scheduled to be deleted, millisecond unit. eg. 1659519772925
+ */
+const listSecrets = async (res, appid, DB, payload) => {
```

Comment:
deleted secrets will also be listed, right? And whether being deleted cannot be distinguished?

Reply:
Yes, deleted secrets also listed, the plandeletetime is not empty, it is deleted